onedal/neighbors/neighbors.py

```diff
@@ -0,0 +1,369 @@
+#===============================================================================
+# Copyright 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#===============================================================================
+
+from sklearn.utils.multiclass import check_classification_targets
+from sklearn.base import is_classifier, is_regressor
+from abc import ABCMeta, abstractmethod
+from enum import Enum
+import sys
+from numbers import Number, Integral
+import warnings
+
+from distutils.version import LooseVersion
+from sklearn import __version__ as sklearn_version
+
+if LooseVersion(sklearn_version) >= LooseVersion("0.22"):

```

Comment:
this must be in scikit-learn-intelex. onedal - without dispatching of sklearn/onedal! only call to onedal